constraint-solver/src/symbolic_expression.rs

```diff
@@ -508,6 +484,28 @@ impl<T: FieldElement, V: Clone + Eq> RuntimeConstant for SymbolicExpression<T, V
         }
     }
 
+    /// Returns the multiplicative inverse in the field.
+    fn field_inverse(&self) -> Self {
+        if let SymbolicExpression::Concrete(x) = self {
+            assert!(x != &T::from(0));
+            SymbolicExpression::Concrete(T::from(1) / *x)
+        } else if let SymbolicExpression::BinaryOperation(x, BinaryOperator::Div, y, _) = self {
+            SymbolicExpression::BinaryOperation(
+                y.clone(),
+                BinaryOperator::Div,
+                x.clone(),
+                Default::default(),
+            )
+        } else {
+            SymbolicExpression::BinaryOperation(
+                Arc::new(Self::from(T::from(1))),
+                BinaryOperator::Div,
+                Arc::new(self.clone()),
+                Default::default(),
+            )
+        }
+    }
+
```

Comment:
It seems like this is just moved from an impl to a trait impl, but not using the changes in `runtime_constant`, correct?

Reply:
Yes exactly, I used specialized impls for SymbolicExpression because I was not sure if it would simplify correctly if this is reduced to multiplying with the inverse.